map_view/display-device-location-with-nmea-data-sources/src/main/java/com/esri/samples/display_device_location_with_nmea_data_sources/DisplayDeviceLocationWithNmeaDataSourcesController.java

```diff
@@ -0,0 +1,203 @@
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a map with the navigation basemap style and set it to the map view
+      ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_NAVIGATION);
+      mapView.setMap(map);
+
+      // set a viewpoint on the map view centered on Redlands, California
+      mapView.setViewpoint(new Viewpoint(new Point(-117.191, 34.0306, SpatialReferences.getWgs84()), 100000));
+
+      // create a new NMEA location data source
+      nmeaLocationDataSource = new NmeaLocationDataSource(SpatialReferences.getWgs84());
+      // set the NMEA location data source onto the map view's location display
+      LocationDisplay locationDisplay = mapView.getLocationDisplay();
+      locationDisplay.setLocationDataSource(nmeaLocationDataSource);
+      locationDisplay.setAutoPanMode(LocationDisplay.AutoPanMode.RECENTER);
+      // disable map view interaction, the location display will automatically center on the mock device location
+      mapView.setEnableMousePan(false);
+      mapView.setEnableKeyboardNavigation(false);
+      stopButton.setDisable(true);
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Initializes the location data source, reads the mock data NMEA sentences, and displays location updates from that file
+   * on the location display. Data is pushed to the data source using a time line to simulate live updates, as they would
+   * appear if using real-time data from a GPS dongle.
+   */
+  @FXML
+  private void start () {
+
+    // initialize the location data source and prepare to begin receiving location updates when data is pushed. As
+    // updates are received, they will be displayed on the map
+    nmeaLocationDataSource.startAsync();
+
+    // prepare the mock data NMEA sentences
+    File simulatedNmeaDataFile = new File(System.getProperty("data.dir"), "./samples-data/redlands/Redlands.nmea");
+    if (simulatedNmeaDataFile.exists()) {
+
+      try  {
+        // read the nmea file contents using a buffered reader and store the mock data sentences in a list
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(simulatedNmeaDataFile.getPath()));
+        String nmeaSentence;
+        List<String> nmeaSentences = new ArrayList<>();
+
+        while ((nmeaSentence = bufferedReader.readLine()) != null) {
+          // add carriage return for NMEA location data source parser
+          nmeaSentences.add(nmeaSentence + "\n");
+        }
+        // close the stream and release resources
+        bufferedReader.close();
+
+        // add a satellite changed listener to the NMEA location data source and display satellite information on the app
+        setupSatelliteChangedListener();
+
+        // create a new time line to push the mock data NMEA sentences into the data source every 250 ms
```

Comment:
```suggestion
        // create a new timeline to push the mock data NMEA sentences into the data source every 250 ms
```

At least I think this is a correct correction